packages/ifpack2/src/Ifpack2_LocalSparseTriangularSolver_def.hpp

```diff
@@ -398,6 +398,33 @@ compute ()
   if (Teuchos::nonnull (htsImpl_))
     htsImpl_->compute (*A_crs_, out_);
 
+  // non-completed code to reverse the storage order of the local matrix
+  typedef  Tpetra::CrsMatrix<scalar_type,local_ordinal_type,global_ordinal_type,node_type> Matrix;
+  typename Matrix::local_matrix_type A = A_crs_->getLocalMatrix();
+  typename Matrix::local_matrix_type::row_map_type ptr = A.graph.row_map;
+  typename Matrix::local_matrix_type::index_type   ind = A.graph.entries;
+  typename Matrix::local_matrix_type::values_type  val = A.values;
+
+  const local_ordinal_type numRows = A.numRows ();
+
+  typename Matrix::local_matrix_type::row_map_type newptr = ptr;
+  typename Matrix::local_matrix_type::index_type   newind = ind;
+  typename Matrix::local_matrix_type::values_type  newval = val;
+
```

Comment:
Also, you need to allocate newind and newval, not just have them be shallow copies.
```
typename Matrix::local_matrix_type::index_type::non_const_type newind ("ind", ind.dimension_0 ());
typename Matrix::local_matrix_type::values_type newval ("val", val.dimension_0 ());
```